lib/spack/spack/build_systems/nmake.py

```diff
@@ -30,73 +30,117 @@ class NMakePackage(spack.package_base.PackageBase):
-            | :py:attr:`~.NMakeBuilder.install_targets`  | Specify ``nmake``  |
-            |                                            | targets for the    |
-            |                                            | install phase      |
-            +--------------------------------------------+--------------------+
-            | :py:meth:`~.NMakeBuilder.build_directory`  | Directory where the|
-            |                                            | Makefile is located|
-            +--------------------------------------------+--------------------+
+        +-----------------------------------------------+---------------------+
+        | **Method**                                    | **Purpose**         |
+        +===============================================+=====================+
+        | :py:attr:`~.NMakeBuilder.build_targets`       | Specify ``nmake``   |
+        |                                               | targets for the     |
+        |                                               | build phase         |
+        +-----------------------------------------------+---------------------+
+        | :py:attr:`~.NMakeBuilder.install_targets`     | Specify ``nmake``   |
+        |                                               | targets for the     |
+        |                                               | install phase       |
+        +-----------------------------------------------+---------------------+
+        | :py:meth:`~.NMakeBuilder.build_directory`     | Directory where the |
+        |                                               | project makefile    |
+        |                                               | is located          |
+        +-----------------------------------------------+---------------------+
     """
 
-    phases = ("edit", "build", "install")
-
-    #: Names associated with package methods in the old build-system format
-    legacy_methods = ("check", "installcheck")
-
-    #: Names associated with package attributes in the old build-system format
-    legacy_attributes = (
-        "build_targets",
-        "install_targets",
-        "build_time_test_callbacks",
-        "install_time_test_callbacks",
-        "build_directory",
-    )
+    phases = ("build", "install")
 
     #: Targets for ``make`` during the :py:meth:`~.NMakeBuilder.build` phase
     build_targets: List[str] = []
     #: Targets for ``make`` during the :py:meth:`~.NMakeBuilder.install` phase
-    install_targets = ["install"]
-
-    #: Callback names for build-time test
-    build_time_test_callbacks = ["check"]
+    install_targets: List[str] = ["INSTALL"]
 
-    #: Callback names for install-time test
-    install_time_test_callbacks = ["installcheck"]
+    @property
+    def ignore_quotes(self):
+        """Returns boolean representing whether arguments to nmake
+        should have quotes ignored. Default False"""
+        return False
```

Comment:
(question) what is the purpose of ignoring quotes?

Generally we encourage packages to specify options without quotes, and ignore_quotes is true for `Executable`. In other words I suspect this shouldn't be exposed as a public method on this class (and may not need to exist as a method at all on the class).

Reply:
There are a number of cases where packages need custom quoting behavior to perform correctly, particularly in the case where there may be a path with a space specified on the command line. In those instances it's necessary to allow packages to specify the way they handle this, and I thought it would be cleaner, if that's the case, for a package to specify the arguments to nmake as they want and to then handle the quoting with that method rather than overriding the entire install/build methods to set that one flag. Perl is a good example of this.

Reply:
> particularly in the case where there may be a path with a space specified on the command line.

Are there other cases? It might be preferable to register these arguments as paths and have the `Executable` object automatically except such arguments from examination on Windows. Basically: if there's a policy we can extract into the common-use objects I want to avoid users having to consider the same problem repeatedly.

Reply:
Other cases are mostly related to handling issues with ill formed nmake or msbuild files that don't process CLI arguments very well and rely on there being quotes in the argument passed in for various reasons (I recall at least one package like this).  

The issue with abstracting this away from user handling is the use cases are fairly niche, inconsistent, and unpredictable. That being said, I don't really like handling this as a property either, maybe a class variable?

Reply:
I think this should

* keep the builder property (no need to make it a class variable)
* rephrase the docstring for `ignore_quotes` to explain that this controls whether we warn about the presence of quotes, like `Executable` (by itself, I think the meaning of "ignoring quotes" is confusing: e.g. someone might take that to mean that we discard them)
* also mention in the docstring that builders for some packages may want to set `ignore_quotes=True` when the builder in fact needs to quote some arguments

Overall then build systems would set ignore_quotes=True (to mean that we shouldn't complain about the presence of quotes) if:

* Paths contain spaces (specifically paths passed as arguments to the builder exe)
* The build script parsing expects quotes

Reply:
Sounds good, will amend now.

Reply:
Amended and pushed.